src/Dfc.CourseDirectory.Core/Search/AzureSearch/SearchTextExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Dfc.CourseDirectory.Core.Search.AzureSearch
+{
+    public static class SearchTextExtensions
+    {
+        public static string EscapeSimpleQuerySearchOperators(this string searchText)
+        {
+            if (searchText == null)
+            {
+                return string.Empty;
+            }
+
+            return Regex.Replace(searchText, @"([\\\+\|\""\(\)\'\-\*\?])", "\\$1")
+                .Replace("#", string.Empty); // Appears to intefere with wildcarding in the search analyzer, so remove.
+        }
+
+        public static string AppendWildcardWhenLastCharIsAlphanumeric(this string searchText)
+        {
+            var lastChar = searchText?.LastOrDefault();
+
+            return lastChar.HasValue && char.IsLetterOrDigit(lastChar.Value)
+                ? $"{searchText}*"
+                : searchText ?? string.Empty;
+        }
+
+        public static string RemoveNonAlphanumericChars(this string searchText)
+        {
+            return new string(searchText?.Where(c => char.IsLetterOrDigit(c) || char.IsWhiteSpace(c)).ToArray());
```

Comment:
Possible NRE if `searchText` is null?

Reply:
Surprisingly not... https://github.com/SkillsFundingAgency/dfc-coursedirectory/pull/1674/files#diff-78e70215a6226e5ddebb1d70fb407ec8R80